File: LongestPalindrome.java
//https://leetcode.com/problems/longest-palindrome/

class Solution {
    public int longestPalindrome(String s) {
        Map<Character, Integer> map = new HashMap<>();

        int odd = 0;
        for(int i = 0; i < s.length(); i++){
            map.put(s.charAt(i), map.getOrDefault(s.charAt(i), 0)+1);
            if(map.get(s.charAt(i))%2!=0){
                odd++;
            }else{
                odd--;
            }
        }

        if(odd>0){
            return s.length()-odd+1;
        }
        return s.length();

    }
}

File: Easy/LongestPalindrome/LongestPalindrome.py
#https://leetcode.com/problems/longest-palindrome/

class Solution:
    def longestPalindrome(self, s: str) -> int:
        if len(s)<=1:
            return len(s)
        
        map = Counter(s)
        total = 0
        containsOdd = False
        for key in map:
            total += map[key] - map[key]%2
            if map[key]%2==1:
                containsOdd=True

        return total+(0^containsOdd)
